script.js

```diff
@@ -1,4 +1,122 @@
+/*
+- player 1 rolls two dice and choose the order of the dice
+- player 2 rolls two dice and choose the order of the dice
+- compare player 1 and player 2 number, state the winner
+- reset the game
+*/
+
+var gameState = "player1Roll";
+var player1RollRecords = [];
+var player2RollRecords = [];
+var player1PointsRecord = [];
+var player2PointsRecord = [];
+
+// roll dice function
+var rollDice = function () {
+  let randomDiceNum = Math.floor(Math.random() * 6) + 1;
+  //console.log(randomDiceNum);
+  return randomDiceNum;
+};
+
+// player1 choose value function
+var chooseNumberFunction1 = function (input) {
+  if (input != 1 && input != 2) {
+    outputMessage = `Player 1, invalid option. 
+      For option 1, please key 1.
+      For option 2, please key 2.`;
+  }
+  if (input == 1) {
+    var player1Points =
+      String(player1RollRecords[0]) + String(player1RollRecords[1]);
+    player1PointsRecord.push(Number(player1Points));
+    outputMessage = `Player 1, your value is ${player1Points}`;
+    gameState = "player2Roll";
+  }
+  if (input == 2) {
+    var player1Points =
+      String(player1RollRecords[1]) + String(player1RollRecords[0]);
+    player1PointsRecord.push(Number(player1Points));
+    outputMessage = `Player 1, your value is ${player1Points}`;
+    gameState = "player2Roll";
+  }
+  return outputMessage;
+};
+
+// player2 choose value function
+var chooseNumberFunction2 = function (input) {
+  if (input != 1 && input != 2) {
+    outputMessage = `Player 2, invalid option. 
+      For option 1, please key 1.
+      For option 2, please key 2.`;
+  }
+  if (input == 1) {
+    var player2Points =
+      String(player2RollRecords[0]) + String(player2RollRecords[1]);
+    player2PointsRecord.push(Number(player2Points));
+    outputMessage = `Player 2, your value is ${player2Points}`;
+    gameState = "compareValues";
+  }
+  if (input == 2) {
+    var player2Points =
+      String(player2RollRecords[1]) + String(player2RollRecords[0]);
+    player2PointsRecord.push(Number(player2Points));
+    outputMessage = `Player 2, your value is ${player2Points}`;
+    gameState = "compareValues";
+  }
+  return outputMessage;
+};
```

Comment:
this whole function is repeated from `chooseNumberFunction1`, can we refactor it such that we have just one common function for both players to choose their numbers?

many ways to do so but one way based on how you've designed your game/code:
you can do a check on whose turn it is by seeing game state is `player1ChooseNumber` or `player2ChooseNumber` then based on this, you can update the correct variables (i.e. player1PointsRecord or player2PointsRecord, etc)